### PR TITLE
optimise compile error message in jdk compiler.(#8942)

### PR DESCRIPTION
**Pull request by vio-lin**
## What is the purpose of the change
optimise jdk compile error message

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [x] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [x] GitHub Actions works fine on your own branch.
- [x] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/test/java/org/apache/dubbo/common/compiler/support/JdkCompilerTest.java [hunk 1]
```diff
@@ -84,4 +88,17 @@ public void test_compileJavaClass1_java8() throws Exception {
             Assertions.assertEquals("Hello world!", sayHello.invoke(instance));
         });
     }
+
+    @Test
+    public void test_getCompileErrorMessage() {
```
Comment (changfubai) [inline dubbo-common/src/test/java/org/apache/dubbo/common/compiler/support/JdkCompilerTest.java:93]: 驼峰命名

File: dubbo-common/src/test/java/org/apache/dubbo/common/compiler/support/JavaCodeTest.java [hunk 2]
```diff
@@ -98,4 +98,17 @@ String getSimpleCodeWithWithExtends() {
         code.append("}");
         return code.toString();
     }
+
+    String getSimpleCodeWithError() {
+        StringBuilder code = new StringBuilder();
+        code.append("package org.apache.dubbo.common.compiler.support;");
+
+        code.append("public class HelloServiceImpl" + SUBFIX.getAndIncrement() + " implements HelloService {");
```
Comment (BurningCN) [inline dubbo-common/src/test/java/org/apache/dubbo/common/compiler/support/JavaCodeTest.java:106]: Fix typo by the way(SUFFIX)

File: dubbo-common/src/test/java/org/apache/dubbo/common/compiler/support/JdkCompilerTest.java (hunk 1 above)
Comment (vio-lin) [reply dubbo-common/src/test/java/org/apache/dubbo/common/compiler/support/JdkCompilerTest.java]: DONE

File: dubbo-common/src/test/java/org/apache/dubbo/common/compiler/support/JavaCodeTest.java (hunk 2 above)
Comment (vio-lin) [reply dubbo-common/src/test/java/org/apache/dubbo/common/compiler/support/JavaCodeTest.java]: DONE

**Review by BurningCN — APPROVED** [review]
LGTM

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**